internal/config/identity/ldap/config-helper.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package ldap
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+)
+
+// ConfigValidator - contains user provided inputs for configuring LDAP in
+// MinIO. These are same as in the Config type. This type is used to provide
+// feedback to debug and fix configuration problems.
+type ConfigValidator struct {
+	ServerAddr     string
+	TLSSkipVerify  bool
+	ServerInsecure bool
+	ServerStartTLS bool
+
+	// Lookup bind LDAP service account
+	LookupBindDN       string
+	LookupBindPassword string
+
+	// User DN search parameters
+	UserDNSearchBaseDistName string
+	UserDNSearchFilter       string
+
+	// Group search parameters
+	GroupSearchBaseDistName string
+	GroupSearchFilter       string
+
+	rootCAs *x509.CertPool
+}
```

Comment:
@donatello can this not be the same config as `type Config` - do we really need a new struct?